vulnerabilities/improve_runner.py

```diff
@@ -218,3 +221,91 @@ def get_or_create_vulnerability_and_aliases(vulnerability_id, alias_names, summa
         logger.info(f"New alias for {vulnerability!r}: {alias_name}")
 
     return vulnerability
+
+
+def get_or_create_vulnerability_without_aliases(inference):
+    """
+    Get or create vulnerabilitiy without aliases
+
+    Try to get vulnerability by matching references,
+    summary and packages. If no vulnerability is found,
+    create a new one. We do not create a new vulnerability
+    only and only if there is an exact existing vulnerability.
+    """
+
+    refs_are_exact_match = True
+
+    vuln_by_refs = {}
+
+    ref_urls_in_inference = sorted([ref.url for ref in inference.references])
+
+    references = VulnerabilityReference.objects.filter(url__in=ref_urls_in_inference)
+
+    if references.count() != len(ref_urls_in_inference):
+        refs_are_exact_match = False
+    else:
+        for ref in references:
+            # saving it as a set to use intersection later
+            vuln_by_refs[ref.url] = set(ref.vulnerabilities.all())
+
+    if refs_are_exact_match:
+        list_of_vulns_by_each_ref = vuln_by_refs.values()
+        # Find all common vulnerabilities amongst all references
+        common_vulns_in_refs = list(set.intersection(*list_of_vulns_by_each_ref))
+        # For a single common vulnerability that have same exact reference
+        # return it
+        for vuln in common_vulns_in_refs:
+            refs_in_vulnerability = Vulnerability.objects.get(
+                vulnerability_id=vuln
+            ).references.all()
+            ref_urls_in_vuln = sorted([ref.url for ref in refs_in_vulnerability])
+            if ref_urls_in_vuln == ref_urls_in_inference:
+                return vuln
+
+        else:
+            # If there are no common vulnerabilities, fall back to creating
+            # a new vulnerability
+            pass
+
+    # This is fallback for cases when we cannot find vulnerability
+    # by references, summary and packages
+    vulnerability = Vulnerability(summary=inference.summary)
+    vulnerability.save()
+
+    return vulnerability
+
+
+# def match_packages(inference, vuln):
```

Comment:
Please remove commented code.

Reply:
Yes this is not ready for review as of now. It's out of scope for now ( this needs model changes, which we will consider after the migration of importers is done )